src/main/java/org/codehaus/mojo/license/utils/UrlRequester.java

```diff
@@ -82,80 +82,85 @@ public static boolean isStringUrl( String data )
+    {
         // by RFC url is composed by <protocol>:<schema-part>.
         // Here could not be used the URL parser because classpath does not have a registered Handler
         String protocol = StringUtils.substringBefore( url, ":" ).toLowerCase();
-        Charset utf8 = Charset.forName( "UTF-8" );
+        Charset charset = Charset.forName( encoding );
 
         String result = null;
-        try
+        if ( CLASSPATH_PROTOCOL.equals( protocol ) )
         {
-            if ( CLASSPATH_PROTOCOL.equals( protocol ) )
+            ClassLoader classLoader = UrlRequester.class.getClassLoader();
+
+            String resource = url.substring( CLASSPATH_PROTOCOL.length() + 1 );
+            URL resourceUrl = classLoader.getResource( resource );
+            if ( resourceUrl == null )
             {
-                ClassLoader classLoader = UrlRequester.class.getClassLoader();
-    
-                String resource = url.substring( CLASSPATH_PROTOCOL.length() + 1 );
-                URL resourceUrl = classLoader.getResource( resource );
-                if ( resourceUrl == null )
-                {
-                    resourceUrl = ClassLoader.getSystemResource( resource );
-                }
-                if ( resourceUrl != null )
-                {
-                    result = IOUtils.toString( resourceUrl, utf8 );
-                }
-                else
-                {
-                    throw new MojoExecutionException( "The resource " + resource
-                            + " was not found in the maven plugin classpath" );
-                }
+                resourceUrl = ClassLoader.getSystemResource( resource );
             }
-            else if ( "http".equals( protocol ) || "https".equals( protocol ) )
+            if ( resourceUrl != null )
             {
-                CloseableHttpClient httpClient = HttpClientBuilder.create().build();
-                HttpGet get = new HttpGet( url );
-                CloseableHttpResponse response = null;
-                try
-                {
-                    response = httpClient.execute( get );
-                    result = IOUtils.toString( response.getEntity().getContent(), utf8 );
-                }
-                catch ( ClientProtocolException e )
-                {
-                    throw new MojoExecutionException( e.getMessage(), e );
-                }
-                finally
+                result = IOUtils.toString( resourceUrl, charset );
+            }
+            else
+            {
+                throw new IOException( "The resource " + resource
+                        + " was not found in the maven plugin classpath" );
+            }
+        }
+        else if ( "http".equals( protocol ) || "https".equals( protocol ) )
+        {
+            CloseableHttpClient httpClient = HttpClientBuilder.create().build();
+            HttpGet get = new HttpGet( url );
+            CloseableHttpResponse response = null;
```

Comment:
Our baseline is Java 7. Plz use `try (CloseableHttpResponse response  = httpClient.execute( get ))` to have less lines.